examples/libadwaita/tab_game.rs

```diff
@@ -254,17 +254,21 @@ impl Component for AppModel {
             counters: FactoryVecDeque::new(widgets.tabs.clone(), &sender.input),
             start_index: None,
         };
+
+        let mut counters_guard = model.counters.guard();
         for i in 0..3 {
-            model.counters.push_back(i);
+            counters_guard.push_back(i);
         }
-        model.counters.render_changes();
+
+        // Explicitly drop the guard,
+        // so that 'model' is no longer borrowed
+        // and can be moved inside ComponentParts
+        drop(counters_guard);
```

Comment:
The drop here is indeed not the nicest solution. It should work with scopes too. Can you try wrapping the "guarded" part into brackets {} ?

Reply:
Makes sense, but I think with scopes it's still confusing.

Reply:
It's not an uncommon pattern to do something like this in Rust. I know it's not the prettiest thing, but IMO still better than dropping manually.

```Rust
let mutex = Mutex::new();
// Critical section
{
    let guard = mutex.lock();
    // Do critical stuff
} // Make sure the guard is released
// Do non critical stuff
```